### PR TITLE
hbase: various fixes and improvements

### PR DESCRIPTION
**Pull request by UniqMartin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

A bunch of adjustments, the most important being reduced pollution of `bin` and a relocated logs directory. (See individual commit messages and comments in commit diffs for a full list.)

cc @CamJN as you seem to be much more familiar with HBase than I am. It would be much appreciated if you could check whether some of those changes make sense or whether some of them need to be tweaked (e.g. what should be available in `bin`). Thanks!

### CODE REVIEW

**Comment by DomT4** [comment]
LGTM pending @CamJN's 👍. Agree with your notations about removing generic executables from the `$PATH`.

**Comment by CamJN** [comment]
👍 here.

**Comment by UniqMartin** [comment]
Thanks for your input @CamJN and @DomT4!

----

For documentation purposes, here's the list of binaries still available via `bin/` as of this PR:

```
hbase
start-hbase.sh
stop-hbase.sh
```

The ones that will no longer make it into PATH for direct execution (still accessible via `libexec/bin/`):

```
draining_servers.rb
get-active-master.rb
graceful_stop.sh
hbase-cleanup.sh
hbase-common.sh
hbase-config.sh
hbase-daemon.sh
hbase-daemons.sh
hbase-jruby
hirb.rb
local-master-backup.sh
local-regionservers.sh
master-backup.sh
region_mover.rb
region_status.rb
regionservers.sh
replication
rolling-restart.sh
shutdown_regionserver.rb
test
thread-pool.rb
zookeepers.sh
```

(Fun little detail: `replication` and `test` are actually directories in `libexec/bin/`, but the current bottle has `exec` shell wrappers in `bin/` for them nonetheless.)

File: Formula/hbase.rb [hunk 1]
```diff
@@ -55,6 +61,11 @@ def install
               "export HBASE_OPTS=\"-Djava.net.preferIPv4Stack=true -XX:+UseConcMarkSweepGC\"")
       s.gsub!("# export JAVA_HOME=/usr/java/jdk1.6.0/",
               "export JAVA_HOME=\"$(/usr/libexec/java_home)\"")
+
+      # Default `$HBASE_HOME/logs` is unsuitable as it would cause writes to the
+      # formula's prefix.
+      s.gsub!(/^# export HBASE_LOG_DIR=.*$/,
+              "export HBASE_LOG_DIR=#{var}/log/hbase")
```
Comment (UniqMartin) [inline Formula/hbase.rb]: I just realized this also prevents legitimate overrides of this variable, e.g. in the [test of the `hbase` formula](https://github.com/Homebrew/homebrew-core/blob/b9c7d819782ac7062273111ae7cf4d48e647b4c7/Formula/hbase.rb#L156). Will be tweaking this a bit to allow for this kind of customization.

**Comment by MikeMcQuaid** [comment]
👍 🚢

**Pull request closed without merging**